packages/react/src/Grid/Grid.tsx

```diff
@@ -99,6 +99,41 @@ export const Grid = forwardRef(
   ),
 ) as GridComponent
 
+type GridCellFullWidthProp = {
+  /** Whether the cell spans the full width of the grid. */
+  fullWidth?: boolean
+  span?: never
+  start?: never
+}
+
+type GridCellColumnProps = {
+  fullWidth?: never
+  /** The amount of grid columns the cell spans. */
+  span?: GridColumnNumber | GridColumnNumbers
+  /** The index of the grid column the cell starts at. */
+  start?: GridColumnNumber | GridColumnNumbers
+}
+
+export type GridCellProps = (GridCellFullWidthProp | GridCellColumnProps) &
+  PropsWithChildren<HTMLAttributes<HTMLDivElement>>
+
+const GridCell = forwardRef(
```

Comment:
Can you add tests for the `Grid` and `GridCell` components? If that's out of scope for this PR, we could do it in a seperate ticket

Reply:
Ah, sorry, missed that somehow